backend/penndata/management/commands/get_events.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+from urllib.parse import urljoin
+
+from bs4 import BeautifulSoup
+from django.core.management.base import BaseCommand
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.ui import WebDriverWait
+
+from penndata.models import Event
+
+
+PENN_TODAY_WEBSITE = "https://penntoday.upenn.edu/events"
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **kwargs):
+        now = datetime.datetime.now()
+        current_month = now.month
+        current_year = now.year
+
+        # Clears out previous Events
+        # past_events = Event.objects.filter(end__lt=now.date())
+        # past_events.delete()
+
+        # Scrapes Penn Today
+        try:
+            driver = webdriver.Chrome()
+
+            driver.get(PENN_TODAY_WEBSITE)
+            events_list = WebDriverWait(driver, 10).until(
+                EC.presence_of_element_located((By.ID, "events-list"))
+            )
+
+            html_content = events_list.get_attribute("innerHTML")
+            driver.quit()
+        except ConnectionError:
+            return None
+
+        soup = BeautifulSoup(html_content, "html.parser")
+
+        event_articles = soup.find_all("article", class_="tease")
+
+        for article in event_articles:
+            name = article.find("h3", class_="tease__head").text.strip()
+            description = article.find("div", class_="tease__dek").text.strip()
+            start_date_str = article.find("p", class_="tease__date").text.strip()
+            meta_elements = article.find_all("p", class_="tease__meta--sm")
+            if len(meta_elements) >= 2:
+                start_time_str = meta_elements[0].text.strip().replace(".", "")
+                location = meta_elements[1].text.strip()
+            else:
+                start_time_str = "All day"
+                location = None
+
+            end_date_elem = article.find(
+                "p", class_="tease__meta--sm", string=lambda x: "Through" in str(x)
+            )
+            end_date = None
+            if end_date_elem:
+                end_date_str = end_date_elem.text.strip().split(" ")[-1]
+                end_date = (
+                    datetime.datetime.strptime(end_date_str, "%m/%d/%Y")
+                    + datetime.timedelta(days=1)
+                    - datetime.timedelta(seconds=1)
+                )
+
+            if start_date_str == "02/29" or start_date_str == "2/29":
```

Comment:
Any reason we needed to split the logic up if it is leap year?

Reply:
some issue happens with the year because we handle that separately, since the date doesn't automatically include year. So, if it's a leap year, it won't process until after the string date is turned into month day, which will make it break